### ISSUE TITLE
REQUEST

### ISSUE THREAD

**Issue by ConfidentialCode** [TheUnknownCod3r/BO4-Lucy-Menu#28]
Hi, Wonderfall job on the menu!!!

I'd like to request 2 things.

1. Unlimited specialist and instant level 3 specialist weapon 

2. Unlock PAP

Thanks

**Comment by TheUnknownCod3r** [comment]
Hey, I appreciate the feedback! In regards to your suggestions, I can definitely do the first, you can already get your lv3 specialist in the weapons menu, and can choose which one you'd like, but I can definitely do infinite specialist.

As for the second, thats a little harder. In order to open the PAP correctly, on some maps it requires connecting paths, triggering specific flags, and using specific functions per map. At least, this is my current knowledge of BO4 and the way things are threaded. If I was to add this for a map (Say example, IX), this would limit support to that map only. If I can figure out a way in future to correctly call this, It will be added, but for now its out of my knowledge. You can however pack a punch your current weapon via the menu, or use open all doors / debris and then do it manually.

**Comment by ConfidentialCode** [comment]
Awesome, I understand the PAP and how it would be difficult to execute given the different map and steps to unlock. Thanks again for updating Lucy!!!

**Comment by NotNierPea** [comment]
> 
> As for the second, thats a little harder. In order to open the PAP correctly, on some maps it requires connecting paths, triggering specific flags, and using specific functions per map. At least, this is my current knowledge of BO4 and the way things are threaded. If I was to add this for a map (Say example, IX), this would limit support to that map only. If I can figure out a way in future to correctly call this, It will be added, but for now its out of my knowledge. You can however pack a punch your current weapon via the menu, or use open all doors / debris and then do it manually.

there is a way to do functions with lazy link -> (without crashing in other maps), there is one guy named ATE47, and he made support for lazy link and detours etc, for bo4, you will have to use his special compiler from serious in his GitHub, it should pretty fine for all maps (i tested it too) 

OR, you can also use gamesettings for papenabled or flags like you said. (i made a lot of mods for bo4 and i have experience in it lol)

Currently for me, and some of my friends, we have found most ways of skipping to EE steps/boss fight too (with some flags and some lazy link)

**Comment by TheUnknownCod3r** [comment]
> > As for the second, thats a little harder. In order to open the PAP correctly, on some maps it requires connecting paths, triggering specific flags, and using specific functions per map. At least, this is my current knowledge of BO4 and the way things are threaded. If I was to add this for a map (Say example, IX), this would limit support to that map only. If I can figure out a way in future to correctly call this, It will be added, but for now its out of my knowledge. You can however pack a punch your current weapon via the menu, or use open all doors / debris and then do it manually.
> 
> there is a way to do functions with lazy link -> (without crashing in other maps), there is one guy named ATE47, and he made support for lazy link and detours etc, for bo4, you will have to use his special compiler from serious in his GitHub, it should pretty fine for all maps (i tested it too)
> 
> OR, you can also use gamesettings for papenabled or flags like you said. (i made a lot of mods for bo4 and i have experience in it lol)
> 
> Currently for me, and some of my friends, we have found most ways of skipping to EE steps/boss fight too (with some flags and some lazy link)
> 
> although detours are not supported for base game, (you will have to use a client named Shield/Project BO4)
> 
> here is some stuff i did with lazy link (:
> 
> ![Screenshot 2023-11-01 011238](https://user-images.githubusercontent.com/130810875/279527109-d6fbecc8-ede7-4e18-80ad-9536957180fb.png)

Yeah see, I would be willing to do that if it didn't break compatibility with the base game. I am aware of how to do it using detours etc, but keeping it compatible with Battle.net is also a priority for me. Maybe if / when Serious updates official support to T7 Compiler, I'll look into it. Until then, if thats all you really want, I believe ate47's Atian menu can do it as well. I'll honestly probably be leaving this as is now except for bug fixes, and recommending people move over to that as / when features are added, or a better alternative comes up.

**Comment by NotNierPea** [comment]
> Yeah see, I would be willing to do that if it didn't break compatibility with the base game. I am aware of how to do it using detours etc, but keeping it compatible with Battle.net is also a priority for me. Maybe if / when Serious updates official support to T7 Compiler, I'll look into it. Until then, if thats all you really want, I believe ate47's Atian menu can do it as well. I'll honestly probably be leaving this as is now except for bug fixes, and recommending people move over to that as / when features are added, or a better alternative comes up.

hey there, i wanted to show your menu being displayed in a custom hud by shield client (thanks to ate47) 
https://youtu.be/VoFo1Ce6Zg8

unfortunately, this only works for host for gsc huds side, csc huds works for other players but its very hard to do so, but at least its not using println to lag the game anymore.

**Comment by TheUnknownCod3r** [comment]
> > Yeah see, I would be willing to do that if it didn't break compatibility with the base game. I am aware of how to do it using detours etc, but keeping it compatible with Battle.net is also a priority for me. Maybe if / when Serious updates official support to T7 Compiler, I'll look into it. Until then, if thats all you really want, I believe ate47's Atian menu can do it as well. I'll honestly probably be leaving this as is now except for bug fixes, and recommending people move over to that as / when features are added, or a better alternative comes up.
> 
> hey there, i wanted to show your menu being displayed in a custom hud by shield client (thanks to ate47) https://youtu.be/VoFo1Ce6Zg8
> 
> unfortunately, this only works for host for gsc huds side, csc huds works for other players but its very hard to do so, but at least its not using println to lag the game anymore.

That is pretty nice! Didn't know that sort of thing was possible at this point. I might have to look at shield

**Comment by TheUnknownCod3r** [comment]
Alright, So I have been looking into switching to Shield for a more expanded version as others have shared above, and while I feel its a worthwhile change to make, I do think it'll mean two separate versions of the menu for each copy of the game. There is already a port of a ton of Lucy features to Shield by @NotNierPea at https://github.com/NotNierPea/RushMod-T8 which I also looked at, so if Nier is happy, I can just refer people to his codebase for Shield, and keep this one for Battle.net Support. Also Nice work!

**Comment by NotNierPea** [comment]
> Alright, So I have been looking into switching to Shield for a more expanded version as others have shared above, and while I feel its a worthwhile change to make, I do think it'll mean two separate versions of the menu for each copy of the game. There is already a port of a ton of Lucy features to Shield by @NotNierPea at https://github.com/NotNierPea/RushMod-T8 which I also looked at, so if Nier is happy, I can just refer people to his codebase for Shield, and keep this one for Battle.net Support. Also Nice work!

Thanks, Maybe I will try to port your menu into shield in a separate mod, and also right now shield has updated and added LUA support, it’s still very hard to deal with, but maybe that can be used for Lucy menu for other players instead of being host only.

**Comment by TheUnknownCod3r** [comment]
LV3 Specialists are added, Once Serious officially adds Detours, I can add the additional features, or I can switch injection to Atian Cod Tools, and use his detours support, which would break t7 support in the meantime. 

Also, Nier, in regards to your comment about Lua for other players instead of host only, the menu is already all client, just stupid iPrintLn. I honestly wish I could just use HudElems like in other games, but just never looked into it enough to know if it was possible on the base game. iirc it is on Shield, but never actually tried it.

**Comment by TheUnknownCod3r** [comment]
Reopening for Logging, Opening Pap is definitely possible, i've just fixed and re-added Gillams original script. Will work on other maps in due time.
